src/main/mondrian/rolap/SqlTupleReader.java

```diff
@@ -841,7 +841,7 @@ private void savePartialResult(List<List<RolapMember>> partialResult) {
                 if (fullyJoiningBaseCubes.size() > 1) {
                     for (int i = 0; i < types.size(); i++) {
                         unionQuery.addOrderBy(
-                            i + 1 + "",
+                            "c" + i,
                             true,
```

Comment:
Yep, I've tried full test suite locally on MySQL 5.7.10 and on MySQL 5.5 on CI-server.